src/panels/lovelace/hui-root.ts

```diff
@@ -110,6 +110,168 @@ class HUIRoot extends LitElement {
+            target="_blank"
+          >
+            <ha-icon-button
+              .label=${this.hass!.localize("ui.panel.lovelace.menu.help")}
+              .path=${mdiHelpCircle}
+            ></ha-icon-button>
+          </a>`
+      );
+    }
+
+    const items: {
+      icon: string;
+      key: LocalizeKeys;
+      overflowAction?: any;
+      buttonAction?: any;
+      visible: boolean | undefined;
+      overflow: boolean;
+      overflow_can_promote?: boolean;
+    }[] = [
+      {
+        icon: mdiFormatListBulletedTriangle,
+        key: "ui.panel.lovelace.unused_entities.title",
+        overflowAction: this._handleUnusedEntities,
+        visible: this._editMode && !__DEMO__,
+        overflow: true,
+      },
+      {
+        icon: mdiCodeBraces,
+        key: "ui.panel.lovelace.editor.menu.raw_editor",
+        overflowAction: this._handleRawEditor,
+        visible: this._editMode,
+        overflow: true,
+      },
+      {
+        icon: mdiViewDashboard,
+        key: "ui.panel.lovelace.editor.menu.manage_dashboards",
+        overflowAction: this._handleManageDashboards,
+        visible: this._editMode && !__DEMO__,
+        overflow: true,
+      },
+      {
+        icon: mdiFileMultiple,
+        key: "ui.panel.lovelace.editor.menu.manage_resources",
+        overflowAction: this._handleManageResources,
+        visible: this._editMode && this.hass.userData?.showAdvanced,
+        overflow: true,
+      },
+      {
+        icon: mdiMagnify,
+        key: "ui.panel.lovelace.menu.search",
+        buttonAction: this._showQuickBar,
+        overflowAction: this._handleShowQuickBar,
+        visible: !this._editMode,
+        overflow: this.narrow,
+      },
+      {
+        icon: mdiCommentProcessingOutline,
+        key: "ui.panel.lovelace.menu.assist",
+        buttonAction: this._showVoiceCommandDialog,
+        overflowAction: this._handleShowVoiceCommandDialog,
+        visible:
+          !this._editMode && this._conversation(this.hass.config.components),
+        overflow: this.narrow,
+      },
+      {
+        icon: mdiRefresh,
+        key: "ui.common.refresh",
+        overflowAction: this._handleRefresh,
+        visible: !this._editMode && this._yamlMode,
+        overflow: true,
+      },
+      {
+        icon: mdiShape,
+        key: "ui.panel.lovelace.unused_entities.title",
+        overflowAction: this._handleUnusedEntities,
+        visible: !this._editMode && this._yamlMode,
+        overflow: true,
+      },
+      {
+        icon: mdiRefresh,
+        key: "ui.panel.lovelace.menu.reload_resources",
+        overflowAction: this._handleReloadResources,
+        visible:
+          !this._editMode &&
+          (this.hass.panels.lovelace?.config as LovelacePanelConfig)?.mode ===
+            "yaml",
+        overflow: true,
+      },
+      {
+        icon: mdiPencil,
+        key: "ui.panel.lovelace.menu.configure_ui",
+        overflowAction: this._handleEnableEditMode,
```

Comment:
Can't we drop `overflowAction` and always guard the `buttonAction` with a `shouldHandleRequestSelectedEvent` when in overflow?

Reply:
Probably, though I'm having a hard time figuring out exactly what this looks like. 

If I try to share one function between both, when I click the overflow menu we get a CustomEvent, and when I click the button I get a PointerEvent. 

If I pass the PointerEvent to `shouldHandleRequestSelectedEvent`, it returns false, and no action occurs.  Should I add some logic to each function to try to differentiate the type of event being passed? Is that better than what we have?

I don't think I understand what the intention of that `shouldHandle` function is, what it is guarding against?